File: Solution/WebSolution/AssessmentPortal/Microservices/APIforSubject/Services/SubjectService.cs
using Transflower.Entities;
using Transflower.SubjectAPI.Repositories.Interfaces;
using Transflower.SubjectAPI.Services.Interfaces;

namespace Transflower.SubjectAPI.Services;

public class SubjectService : ISubjectService
{
   private readonly ISubjectRepository _repository;
   
   public SubjectService(ISubjectRepository repository)
   {
      _repository = repository;

   }
   /*public async Task<SubjectModel> GetDetails(int subjectId)
   {
      return await _repository.GetSubjectById(subjectId);
   }*/
   public async Task<List<SubjectModel>> GetAllSubject()
   {
      return await _repository.GetAllSubject();
   }

   
}

File: Solution/WebSolution/Java/AssessmentAPI/src/main/java/com/transflower/tflassessment/demo/services/SubjectService.java
package com.transflower.tflassessment.demo.services;

import com.transflower.tflassessment.demo.entities.SubjectModel;
import java.util.*;

public interface SubjectService {

    List<SubjectModel> getAllSubjects();

    int addSubject(SubjectModel subject);
    
    int deleteSubject(int subjectId);
}
